src/groupeddataframe/splitapplycombine.jl

```diff
@@ -944,7 +1001,12 @@ function append_rows!(rows, outcols::NTuple{N, AbstractVector},
         cn = colnames[j]
         local vals
         try
-            vals = getproperty(rows, cn)
+            if fcallres isa AbstractVector
```

Comment:
What situation requires this? That makes the design even more complex. :-/

Reply:
This is required because we allow mixing `AbstractVectors` with `AbstractMatrix`/`NamedTuple`/`DataFrame`. Maybe we should really be strict about not changing the return type of functions that are called in aggregations (then the code would be simpler but this would be breaking; though I do not assume anyone does such mixing in practice).

Reply:
Maybe disallow it. I don't really see why somebody would like to do that. How did the old code allow it without adding this check here?

Reply:
It auto-generated default names if needed (:x1, :x2, etc.) and only at the end renamed columns to what was requested (again - if needed). In effect things like https://github.com/JuliaData/DataFrames.jl/issues/2071 were allowed.

Reply:
now we disallow it so this was removed